llvm/lib/Target/DirectX/DirectXPassRegistry.def

```diff
@@ -23,7 +23,10 @@ MODULE_ANALYSIS("dxil-resource-md", DXILResourceMDAnalysis())
 #ifndef MODULE_PASS
 #define MODULE_PASS(NAME, CREATE_PASS)
 #endif
+MODULE_PASS("dxil-intrinsic-expansion", DXILIntrinsicExpansion())
+MODULE_PASS("dxil-op-lower", DXILOpLowering())
+MODULE_PASS("dxil-pretty-printer", DXILPrettyPrinterPass(dbgs()))
+MODULE_PASS("dxil-translate-metadata", DXILTranslateMetadata())
 // TODO: rename to print<foo> after NPM switch
 MODULE_PASS("print-dx-shader-flags", dxil::ShaderFlagsAnalysisPrinter(dbgs()))
-MODULE_PASS("print-dxil-resource-md", DXILResourceMDPrinterPass(dbgs()))
```

Comment:
Should we remove DXILResourceMDPrinterPass in lib\Target\DirectX\DXILResourceAnalysis.cpp ?

Reply:
Yes we should. I currently have that as part of this change that isn't quite ready for review yet: https://github.com/bogner/llvm-project/commit/bed151261c5b26519f6e44c44631cc1f770dfd78

Maybe I should pull the deletion into this PR instead?

Reply:
I've pulled the removal of the pass into this change.